### PR TITLE
Check whether genbank files are gzipped and better handle errors

### PR DESCRIPTION
**Pull request by pkerpedjiev**
## Description

> What was changed in this pull request?

Added a check for the genbank fetcher to see if a file is gzipped and improved the error handling to better inform when something goes wrong.

> Why is it necessary?

1. The genbank fetcher doesn't make the distinction between gzipped and regular content.
2. When errors occur it can be difficult to figure out what went wrong.

Fixes #___

## Checklist

- [x] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [ ] Unit tests added or updated
- [ ] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [ ] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

### CODE REVIEW

File: app/scripts/data-fetchers/genbank-fetcher.js [hunk 1]
```diff
@@ -106,18 +106,22 @@ class GBKDataFetcher {
...
     this.trackUid = slugid.nice();
 
+    const extension = dataConfig.url.slice(dataConfig.url.length - 3);
+    const gzipped = extension === '.gz';
+    this.errorTxt = '';
+
     this.dataPromise = fetch(dataConfig.url, {
       headers: {
-        'Content-Encoding': 'gzip'
+        'Content-Encoding': gzipped ? 'gzip' : 'text/plain',
```
Comment (flekschas) [inline app/scripts/data-fetchers/genbank-fetcher.js]: Just out of curiosity: what if a server automatically gzips the returned plain text. Would this still work?

File: app/scripts/data-fetchers/genbank-fetcher.js [hunk 2]
```diff
@@ -106,18 +106,22 @@ class GBKDataFetcher {
...
     })
-      .then(response => response.arrayBuffer())
+      .then(response => (gzipped ? response.arrayBuffer() : response.text()))
       .then((buffer) => {
-        const gffText = pako.inflate(buffer, { to: 'string' });
+        const gffText = gzipped ? pako.inflate(buffer, { to: 'string' }) : buffer;
         this.gbJson = genbankParser(gffText);
         this.cdss = shuffle(this.gbJson[0]
-          .features.filter(f => f.type === 'gene')
+          .features.filter(f => f.type === 'CDS')
```
Comment (flekschas) [inline app/scripts/data-fetchers/genbank-fetcher.js]: Was this change intended? If so ignore my comment

File: app/scripts/data-fetchers/genbank-fetcher.js (hunk 1 above)
Comment (pkerpedjiev) [reply app/scripts/data-fetchers/genbank-fetcher.js]: That should get decoded by the browser and be returned as plaintext to fetch.
